subprojects/testing-base/src/main/java/org/gradle/api/internal/tasks/testing/worker/TestWorker.java

```diff
@@ -73,8 +73,13 @@ public void execute(final WorkerProcessContext workerProcessContext) {
             }
         } finally {
             LOGGER.info("{} finished executing tests.", workerProcessContext.getDisplayName());
-            // Clean out any security manager the tests might have installed
-            System.setSecurityManager(null);
+
+            try {
+                // Clean out any security manager the tests might have installed
+                System.setSecurityManager(null);
+            } catch (SecurityException e) {
+                LOGGER.info("Unable to clear out SecurityManager, continuing anyway.", e);
```

Comment:
I think we should check whether the `SecurityManager` was changed by the test and only attempt to reset it if it was. If the test did install a `SecurityManager`, we should log a warning "Unable to clear out SecurityManager installed by a test, continuing anyway".